### PR TITLE
[consorbank] Support for older dividend files

### LINKED ISSUE #225: [Consorsbank] Die Divenden können nicht importiert werden
## Document
<details><summary>DIVIDENDENGUTSCHRIFT_838891405_wknA2JNEW_dat20180629_id705708544.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Consorsbank",
    "• 90318 Nürnberg",
    "Jckzw Vhhytpnbu",
    "Obpqzemzgqkh. 5",
    "99458 Püwqgvlw",
    "Depotnummer:",
    "3993040803",
    "7683604064/28",
    "Vermerk der Bank:",
    "1000",
    "2",
    "Datum:",
    "29.06.2018",
    "Seite:",
    "1 von 2",
    "Dividendengutschrift",
    "Wertpapierbezeichnung",
    "WKN",
    "ISIN",
    "Total S.A. Anrechte",
    "A2JNEW",
    "FR0013333374",
    "Bestand",
    "25 Stück",
    "Dividende pro Stück 0,62 EUR",
    "Schlusstag 27.06.2018",
    "Brutto",
    "15,50 EUR",
    "abzgl. Quellensteuer",
    "30,00 % von",
    "15,50 EUR",
    "4,65 EUR",
    "Netto zugunsten",
    "IBAN",
    "DE07 7603 0080",
    "0830 8914 03",
    "10,85 EUR",
    "Valuta 28.06.2018",
    "BIC",
    "CSDBDE71XXX",
    "Anrechenbare Quellensteuer",
    "12,80 % von",
    "15,50 EUR",
    "1,98",
    "EUR",
    "Dividendenzahlung für Extag 11.06.2018"
  ],
  [
    "Details zur Abrechnung",
    "Steuerpflichtiger Gesamtertrag",
    "15,50 EUR",
    "Mit Verrechnungstopf \"Allgemein\" verrechnet",
    "0,00 EUR",
    "Mit Sparerpauschbetrag verrechnet",
    "-15,50 EUR",
    "Mit Quellensteuer verrechnet",
    "0,00 EUR",
    "somit verrechnete Quellensteuer",
    "0,00 EUR",
    "Bemessungsgrundlage für Kapitalertragsteuer gesamt",
    "0,00 EUR",
    "Details zu den Verrechnungstöpfen",
    "Verrechnungstopf \"Allgemein\" vor der Abrechnung",
    "0,00 EUR",
    "Mit Verrechnungstopf \"Allgemein\" verrechnet",
    "0,00 EUR",
    "Verrechnungstopf \"Allgemein\" nach der Abrechnung",
    "0,00 EUR",
    "Sparerpauschbetrag vor der Abrechnung",
    "158,14 EUR",
    "Mit Sparerpauschbetrag verrechnet",
    "-15,50 EUR",
    "Sparerpauschbetrag nach der Abrechnung",
    "142,64 EUR",
    "Verrechnungstopf \"Quellensteuer\" vor der Abrechnung",
    "5,46 EUR",
    "Änderung Verrechnungstopf \"Quellensteuer\"",
    "1,98 EUR",
    "Verrechnungstopf \"Quellensteuer\" nach der Abrechnung",
    "7,44 EUR",
    "Depotnummer:",
    "0838891405",
    "1110496558/00",
    "Vermerk der Bank:",
    "1000",
    "2",
    "Datum:",
    "29.06.2018",
    "Seite:",
    "2 von 2"
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #229: [Consorsbank] Kauf kann nicht importiert werden
## Document
<details><summary>KAUF_838891405_ord91180456_001_wkn251124_dat20150803_id453100601.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland.",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Consorsbank • 90318 Nürnberg",
    "WERTPAPIERABRECHNUNG",
    "KAUF",
    "AM 03.08.2015",
    "UM",
    "08:00:48",
    "MUENCHEN",
    "NR. 91180456.001",
    "Pzihw Whzaogljz",
    "Jfkkiyvffrcl. 0",
    "11828 Rüfcirbw",
    "Hinweis für Kontoauszüge, Wertpapierabrechnungen und Depotbuchungsanzeigen:",
    "Kapitalerträge und Spekulationsgewinne sind einkommensteuerpflichtig.",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung oder die Nichtgenehmigung einer im Wege der Einzugsermächtigung erfolgten",
    "Lastschriftabbuchung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der Allgemeine Geschäftsbedingungen (AGB Banken) sowie",
    "unter B. VII. Ziffer 2.4 der Lastschriftbedingungen. Umsätze und Kontobuchungen, die nach dem Erstellungsdatum anfallen und sich auf den Abrechnungssaldo des",
    "abgelaufenen Abrechnungszeitraumes auswirken, werden erst mit dem folgenden Kontoauszug ausgewiesen. Korrekturen werden seitens der Bank gekennzeichnet.",
    "Machen Sie Ihre Einwendungen in Textform geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank (Revision) oder per Fax oder",
    "Mail an die unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Depotnummer",
    "0556425045",
    "8026412383/53",
    "Vermerk der Bank",
    "1000",
    "00",
    "Wertpapier",
    "WKN",
    "ISIN",
    "ISHS-EO CO.BD LA.C.UTS DZ",
    "251124",
    "DE0002511243",
    "Einheit",
    "Umsatz",
    "ST",
    "0,51764",
    "Kurs",
    "133,240000",
    "EUR",
    "P.ST.",
    "Kurswert",
    "EUR",
    "68,97",
    "Provision",
    "EUR",
    "1,03",
    "Wert 05.08.2015",
    "EUR",
    "70,00",
    "zulasten Konto-Nr. 0830891403",
    "Wertpapiere zugunsten Girosammelverwahrung",
    "Clearstream Bk Ffm 2126",
    "Die Order wurde an folgender Börse gehandelt: MUENCHEN",
    "Sparplanname: Euro Anleihen"
  ],
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland.",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Infos bei Käufen bzw. Verlust bei Wertpapieren ungleich Aktien",
    "Gezahlter Zwischengewinn",
    "EUR",
    "0,49",
    "In den Verrechnungstopf Allgemein eingestellt",
    "EUR",
    "0,49",
    "Summe ergibt",
    "KAPST-pflichtige Kapitalerträge",
    "EUR",
    "0,00",
    "Mit Sparerpauschbetrag verrechnet",
    "EUR",
    "0,00",
    "Bemessungsgrundlage für KAPST vor QUST",
    "EUR",
    "0,00",
    "An Kapitalertrag anrechenbare. ausl. QUST 0,00*4",
    "EUR",
    "0,00",
    "Bemessungsgrundlage für KAPST",
    "EUR",
    "0,00",
    "Bemessungsgrundlage für KAPST anteilig 100,00%",
    "EUR",
    "0,00",
    "Salden nach dem Geschäft",
    "Verrechnungstopf Aktien nach dem Geschäft",
    "EUR",
    "0,00",
    "Verrechnungstopf Allg. nach dem Geschäft",
    "EUR",
    "0,86",
    "Verrechnungstopf QUST nach dem Geschäft",
    "EUR",
    "4,58",
    "Sparerpauschbetrag nach dem",
    "Geschäft",
    "EUR",
    "713,90",
    "Infos zu Investmentpreisen",
    "Akkum. ausschüttungsgleiche Erträge",
    "EUR",
    "4,25",
    "Bereinigte akkum. ausschüttungsgleiche Erträge",
    "EUR",
    "3,58",
    "Akkum. Substanzausschüttungen",
    "EUR",
    "0,08",
    "Seite 2 zu Wertpapierabrechnung Nr.",
    "91180456.001 vom",
    "03.08.2015",
    "Depotnummer 0838891405",
    "Hinweis für Kontoauszüge, Wertpapierabrechnungen und Depotbuchungsanzeigen:",
    "Kapitalerträge und Spekulationsgewinne sind einkommensteuerpflichtig.",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung oder die Nichtgenehmigung einer im Wege der Einzugsermächtigung erfolgten",
    "Lastschriftabbuchung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der Allgemeine Geschäftsbedingungen (AGB Banken) sowie",
    "unter B. VII. Ziffer 2.4 der Lastschriftbedingungen. Umsätze und Kontobuchungen, die nach dem Erstellungsdatum anfallen und sich auf den Abrechnungssaldo des",
    "abgelaufenen Abrechnungszeitraumes auswirken, werden erst mit dem folgenden Kontoauszug ausgewiesen. Korrekturen werden seitens der Bank gekennzeichnet.",
    "Machen Sie Ihre Einwendungen in Textform geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank (Revision) oder per Fax oder",
    "Mail an die unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #231: [Consorsbank] Dividenengutschrift kann nicht übernommen werden
## Document
<details><summary>DIVIDENDENGUTSCHRIFT_838891405_wknA14UJS_dat20150702_id449670684.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland.",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Consorsbank • 90318 Nürnberg",
    "Depotnummer",
    "1212381123",
    "3258608171/14",
    "Vermerk der Bank",
    "1000",
    "00",
    "Ddfqj Yccunpgea",
    "Uqrqmdtftpwk. 7",
    "10773 Püwqmgky",
    "NUERNBERG, 02.07.2015",
    "DIVIDENDENGUTSCHRIFT",
    "ST                       25,00000          WKN:  A14UJS",
    "Total S.A.",
    "Anrechte (Wahldividende)",
    "ZINS-/DIVIDENDENSATZ            0,610000  EUR SCHLUSSTAG PER 30.06.2015",
    "EX-TAG  01.07.2015",
    "BRUTTO                                        EUR                 15,25",
    "QUST 30,00000  %                              EUR                  4,58",
    "WERT 01.07.2015                               EUR                 10,67",
    "ZU GUNSTEN KONTO-NR. 0830 891 403 / IBAN HU56211449541270708838",
    "(BIC CSDBDE71XXX)",
    "ANRECHENBARE AUSLAEND. QUELLENSTEUER          EUR                 2,29",
    "KAPST-PFLICHTIGER KAPITALERTRAG               EUR                15,25",
    "MIT SPARERPAUSCHBETRAG VERRECHNET             EUR                15,25-",
    "BEMESSUNGSGRUNDLAGE FUER KAPST                EUR                 0,00",
    "VERRECHNUNGSTOPF ALLGEMEIN NACH ERTRAG        EUR                 0,00",
    "SPARERPAUSCHBETRAG NACH ERTRAG                EUR               713,90",
    "IN DEN VERRECHNUNGSTOPF QUST EINGESTELLT      EUR                 2,29",
    "VERRECHNUNGSTOPF AUSL. QUST NACH ERTRAG       EUR                 4,58",
    "VOLLER QUST-ABZUG, FUER GEBIETSANSAESSIGE RUECKERSTATTUNG AUF ANTRAG",
    "ES WURDE KEINE KIRCHENSTEUER EINBEHALTEN",
    "JAHRESSTEUERBESCHEINIGUNG FOLGT",
    "KAPITALERTRAEGE SIND EINKOMMENSTEUERPFLICHTIG",
    "DIESE MITTEILUNG WIRD NICHT UNTERSCHRIEBEN"
  ],
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland.",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Hinweis für Zins- und Dividendengutschriften:",
    "Einwendungen",
    "wegen",
    "Unrichtigkeit",
    "oder",
    "Unvollständigkeit",
    "dieser",
    "Mitteilung",
    "müssen",
    "unverzüglich",
    "erhoben",
    "werden,",
    "vgl.",
    "Nummer",
    "B.",
    "Ziffer",
    "I. 11",
    "(4)",
    "und",
    "(5)",
    "der",
    "Allgemeine",
    "Geschäftsbedingungen",
    "(AGB",
    "Banken).",
    "Umsätze",
    "und",
    "Kontobuchungen,",
    "die",
    "nach",
    "dem",
    "Erstellungsdatum",
    "anfallen",
    "und",
    "sich",
    "auf",
    "den",
    "Abrechnungssaldo",
    "des",
    "abgelaufenen",
    "Abrechnungszeitraumes",
    "auswirken,",
    "werden",
    "erst",
    "mit",
    "dem",
    "folgenden",
    "Kontoauszug",
    "ausgewiesen.",
    "Korrekturen",
    "werden",
    "seitens",
    "der",
    "Bank",
    "gekennzeichnet.",
    "Machen Sie Ihre Einwendungen in Textform geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an",
    "die Consorsbank",
    "(Revision)",
    "oder",
    "per",
    "Fax",
    "oder",
    "Mail",
    "an",
    "die",
    "unten",
    "angegebenen",
    "Adressen.",
    "Das",
    "Unterlassen",
    "rechtzeitiger",
    "Einwendungen gilt als Genehmigung.",
    "Die",
    "mit",
    "„Steuerbescheinigung“",
    "gekennzeichneten",
    "Abrechnungen",
    "sind",
    "sorgfältig",
    "aufzubewahren,",
    "da",
    "nur",
    "gegen",
    "Vorlage",
    "dieser",
    "Belege",
    "die",
    "Abrechnung",
    "im",
    "Rahmen",
    "der",
    "steuerlichen",
    "Veranlagung",
    "möglich",
    "ist.",
    "Bei",
    "Stornierung",
    "ist",
    "die",
    "Bank",
    "verpflichtet,",
    "die",
    "falsche",
    "Steuerbescheinigung",
    "zurückzufordern.",
    "Andernfalls",
    "ist",
    "sie",
    "verpflichtet,",
    "nach",
    "Ablauf",
    "eines",
    "Monats",
    "dem",
    "Wohnsitzfinanzamt",
    "des",
    "Depotinhabers",
    "Meldung",
    "zu",
    "machen.",
    "Gebietsansässige",
    "machen",
    "wir",
    "auf",
    "die",
    "ggf.",
    "bestehende",
    "Meldepflicht",
    "an",
    "die",
    "Landeszentralbank",
    "gemäß",
    "Außenwirtschaftsverordnung",
    "aufmerksam.",
    "Für",
    "weitere",
    "Auskünfte",
    "stehen",
    "wir",
    "Ihnen gerne zur Verfügung. Meldevordrucke sind bei uns erhältlich.",
    "Erträge",
    "unterliegen",
    "der",
    "Einkommen-",
    "bzw",
    "Körperschaftssteuer.",
    "Falls",
    "Doppelbesteuerungsabkommen",
    "bestehen,",
    "ist",
    "die",
    "einbehaltene",
    "ausländische",
    "Quellensteuer,",
    "soweit",
    "diese",
    "nicht",
    "erstattungspflichtig",
    "ist,",
    "auf",
    "die",
    "zu",
    "zahlende",
    "Einkommen-",
    "bzw.",
    "Körperschaftssteuer",
    "anrechenbar.",
    "Besteht",
    "mit",
    "dem",
    "Staat,",
    "aus",
    "dem",
    "die",
    "Erträge",
    "zufließen,",
    "kein",
    "Doppelbesteuerungsabkommen",
    "und",
    "werden",
    "ausländische",
    "Steuern",
    "auf",
    "diese",
    "Erträge",
    "einbehalten,",
    "so",
    "ist",
    "die",
    "Steuer",
    "gemäß",
    "§34 c EStG auf die deutsche Einkommen- bzw. Körperschaftssteuer anrechenbar."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #236: Consorsbank - Es können keine Dividendenabrechnungen (älteres Format von 2018) importiert werden
## Document
<details><summary>DIVIDENDENGUTSCHRIFT_718037918_wkn555200_dat20180427_id694030912.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Consorsbank",
    "• 90318 Nürnberg",
    "Rkbavcx Okyio",
    "Vepwevojpi. 49",
    "22152 Ryjlrjkvx",
    "Depotnummer:",
    "9204381066",
    "3103464868/09",
    "Vermerk der Bank:",
    "5640",
    "6",
    "Datum:",
    "27.04.2018",
    "Seite:",
    "1 von 1",
    "Dividendengutschrift",
    "Wertpapierbezeichnung",
    "WKN",
    "ISIN",
    "DEUTSCHE POST AG NAMENS-AKTIEN O.N.",
    "555200",
    "DE0005552004",
    "Bestand",
    "21 Stück",
    "Steuerfreie Dividende pro Stück 1,15 EUR",
    "Schlusstag 24.04.2018",
    "Brutto",
    "24,15 EUR",
    "Netto zugunsten",
    "IBAN",
    "MP74 9018 0667",
    "2156 3204 28",
    "24,15 EUR",
    "Valuta 27.04.2018",
    "BIC",
    "YEVOKK76LMR",
    "FUEHRT GEMAESS BMF-SCHREIBEN VOM",
    "09.10.2012, RANDZIFFOB 93 NET HSXZBKDNNJY DER",
    "ANSCHAFFUNGSKOSTEN."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/5cba3511

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by SirGibihm**
Previously, older dividend documents (prior to 2017) were excluded from import due to missing functions that could parse the data. This was verified by a unittest. I added the necessary functions and removed the old unittest (which didn't work in the first place anyways).
- Fix #231
- Fix #236
- Fix #225
- Fix #229
I also implemented a Fix #225 and while doing so, I replaced most of the tax-calulation logic for dividends which was overly complicated in my eyes. While doing so, I noticed something strange for one of the test-files, i. e. dividend_realty_income.json . Instead of paying the dividend it seems like a previous dividend payout was cancelled, see the last lines that contain both `Netto zulasten` instead of `Netto zugunsten` and `Storno`, see [here](https://github.com/codeofsumit/tresor-import/blob/f500be9f2df8c11b6d4512d279b6daf709cef292/tests/brokers/__mocks__/consorsbank/dividends/dividend_realty_income.json#L48-L62)

So in my eyes, the amount of this file should be negative, as money is taken out of the customers account, am I right? Currently, cancellations are just seen as regular payouts, which is wrong IMO

### CODE REVIEW

**Comment by dredav** [comment]
> So in my eyes, the amount of this file should be negative, as money is taken out of the customers account, am I right?

Oh shit, I did´t see in my PR. Thanks for addressing!

**Review by dredav — APPROVED** [review]
🚀

File: src/brokers/consorsbank.js [hunk 1]
```diff
@@ -8,30 +8,64 @@ import {
...
 
 const findDateDividend = textArr => {
   const keyword = 'valuta';
   const dateLine = textArr.find(t => t.toLowerCase().includes(keyword));
-  return dateLine.substr(keyword.length).trim();
+  if (dateLine !== undefined) {
+    return dateLine.substr(keyword.length).trim();
+  }
+
+  const keywordOld = 'EX-TAG';
```
Comment (dredav) [inline src/brokers/consorsbank.js:64]: I`m not sure if ex-day is the right one. For each old document it seems right so we can use it.

**Review by dredav — APPROVED** [review]

File: src/brokers/consorsbank.js (hunk 1 above)
Comment (SirGibihm) [reply src/brokers/consorsbank.js:64]: > 
> 
> > So in my eyes, the amount of this file should be negative, as money is taken out of the customers account, am I right?
> 
> Oh shit, I did´t see in my PR. Thanks for addressing!

I did not fix this as of yet as I did not know, wether this is what we need and whether a negative amount can be parsed atm. Should I add this?

**Pull request merged**